controls/datagrid/search-as-you-type.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Search as You Type
+page_title: .NET MAUI DataGrid Documentation - Search as You Type
+description: "Learn more about the search as you type feature of the Telerik .NET MAUI DataGrid - how to show/hide the search panel, various configuration settings of the search functionality, methods and events related to searching."
+position: 3
+slug: datagrid-search-as-you-type
+---
+
+# .NET MAUI DataGrid Search as You Type
+
+Telerik .NET MAUI DataGrid provides the ability to search for specific data within its ItemsSource by using its built-in search funcionality. The default behavior is to search as you type, but it can also be performed when the end user finishes typing and presses Enter.
+
+Through the DataGrid's `SearchSettings` property you can control when to show the search panel as well as configure the way the search is performed, so it best suits the end users' needs.
+
+### Show/Hide the Search Panel
+
+The DataGrid's `SearchSettings` exposes  a `SearchPanelVisibilityMode` property which controls when the search panel will appear, here are the available options:
+
+* `NeverVisible`&mdash;the search panel never appears.
+* `ControlledByUser`&mdash;the search panel becomes visible after a user's interaction (pressing Control / Command +F in Windows / MacOS respectively) and its close button is also visible.
+* `AlwaysVisible`&mdash;the search panel is always visible and its close button is not visible.
+
+### Search Panel UI
+
+The search panel provides an entry where the end user can enter the search terms, and also some search options to choose from, such as whether to match the case, whether to filter the results. The additional search settings can be accessed by the Options button (the tree dots) next to the entry.
+
+You can check how the Search Panel UI looks on Windows at the image below:
+
+![Telerik .NET MAUI DataGrid Search Panel](images/datagrid-search-searchpanel.png)
+
+### Search Settings
+
+The default searching behavior works in the following way - all the words divided by space are trimmed from the search text and searched by according to the `LogicalOperator` property.
+
+Below you can find a list of the available configuration options applied through the `RadDataGrid`.`SearchSettings`:
+
+* `IntermediateSearchText`(`string`)&mdash;Specifies the text of the search entry. Change of this value will not necessarily trigger a search, i.e. it will not necessarily change the `SearchText` property. The `SearchText` property is updated in accordance `SearchTrigger` property.
```

Comment:
```suggestion
* `IntermediateSearchText`(`string`)&mdash;Specifies the text of the search entry. A change of this value will not necessarily trigger a search, for example,. it will not necessarily change the `SearchText` property. The `SearchText` property is updated in accordance with the `SearchTrigger` property.
```